broker-daemon/state-machines/order-state-machine.js

```diff
@@ -289,30 +289,31 @@ const OrderStateMachine = StateMachine.factory({
       }
       const dataHandler = ({ orderStatus, fill }) => {
         try {
-          // the Relayer will send a single data message containing the order's state as cancelled and close
-          // the stream if the order has been cancelled. We should handle that and cancel the order locally.
           if (OrderStateMachine.STATES[orderStatus] === OrderStateMachine.STATES.CANCELLED) {
-            this.logger.info(`Order ${orderId} was cancelled on the relayer, cancelling locally.`)
-            return this.tryTo('cancel')
```

Comment:
this is the offending line as it prevents `finish()` from being called, which tears down the listeners.